Bug: CI failing on macOS due to missing timeout command and outdated test paths

## Problem Statement

CI is failing on macOS runners with two issues:

### Issue 1: `timeout` command not found on macOS

The REPL test step fails on macOS with:
```
/Users/runner/work/_temp/c67cbdc2-1aa0-4a27-8272-a592843d865c.sh: line 2: timeout: command not found
```

**Root cause:** The `timeout` command is part of GNU coreutils and is not available by default on macOS. The CI workflow was using `timeout` for the REPL test, which works on Linux but fails on macOS.

### Issue 2: Outdated test file path

The CI log shows:
```
./ez.bin run test/interpreter_tests.ez
Error reading file: open test/interpreter_tests.ez: no such file or directory
```

**Root cause:** An older CI workflow referenced `test/interpreter_tests.ez` (singular `test/`), but the actual test file is at `tests/comprehensive.ez` (plural `tests/`).

## Evidence from CI Logs

From run [#19909699126](https://github.com/SchoolyB/EZ/actions/runs/19909699126):

```
Build and Test (macos-latest, 1.21)  Testing REPL...
Build and Test (macos-latest, 1.21)  /Users/runner/work/_temp/c67cbdc2-1aa0-4a27-8272-a592843d865c.sh: line 2: timeout: command not found
Build and Test (macos-latest, 1.21)  ##[error]Process completed with exit code 1.
```

## Fix Applied

The CI workflow has been updated to handle macOS differently:

**Linux (uses `timeout`):**
```yaml
- name: Test REPL basic functionality (Linux)
  if: runner.os == 'Linux'
  run: |
    echo "Testing REPL..."
    echo -e "temp x int = 5\nx + 10\nexit" | timeout 5 ./ez.bin repl | grep -q "15"
```

**macOS (uses background process with sleep):**
```yaml
- name: Test REPL basic functionality (macOS)
  if: runner.os == 'macOS'
  run: |
    echo "Testing REPL..."
    (echo -e "temp x int = 5\nx + 10\nexit" | ./ez.bin repl > /tmp/repl_output.txt) &
    pid=$!
    sleep 5
    kill $pid 2>/dev/null || true
    grep -q "15" /tmp/repl_output.txt
```

**Test path fixed:**
```yaml
- name: Run interpreter tests
  if: runner.os != 'Windows'
  run: ./ez.bin run tests/comprehensive.ez
```

## Acceptance Criteria

- [ ] CI passes on all platforms (Linux, macOS, Windows)
- [ ] REPL test executes successfully on macOS without `timeout`
- [ ] Comprehensive tests run from correct path `tests/comprehensive.ez`
- [ ] No stale workflow references to `test/interpreter_tests.ez`

## Related Files

- `.github/workflows/ci.yml`
- `tests/comprehensive.ez`

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -34,11 +34,11 @@ jobs:
 
       - name: Run interpreter tests
         if: runner.os != 'Windows'
-        run: ./ez.bin run test/interpreter_tests.ez
+        run: ./ez.bin run tests/comprehensive.ez
 
       - name: Run interpreter tests (Windows)
         if: runner.os == 'Windows'
-        run: .\ez.exe run test\interpreter_tests.ez
+        run: .\ez.exe run tests\comprehensive.ez
 
       - name: Validate example files
         if: runner.os != 'Windows'
@@ -58,11 +58,22 @@ jobs:
             if ($LASTEXITCODE -ne 0) { exit 1 }
           }
 
-      - name: Test REPL basic functionality
-        if: runner.os != 'Windows'
+      - name: Test REPL basic functionality (Linux)
+        if: runner.os == 'Linux'
         run: |
           echo "Testing REPL..."
           echo -e "temp x int = 5\nx + 10\nexit" | timeout 5 ./ez.bin repl | grep -q "15"
+
+      - name: Test REPL basic functionality (macOS)
+        if: runner.os == 'macOS'
+        run: |
+          echo "Testing REPL..."
+          # macOS doesn't have timeout, use a subshell with background process
+          (echo -e "temp x int = 5\nx + 10\nexit" | ./ez.bin repl > /tmp/repl_output.txt) &
+          pid=$!
+          sleep 5
+          kill $pid 2>/dev/null || true
+          grep -q "15" /tmp/repl_output.txt
 
       - name: Test REPL basic functionality (Windows)
         if: runner.os == 'Windows'

